util/stopwaiter/stopwaiter.go

```diff
@@ -179,26 +179,40 @@ func (s *StopWaiterSafe) LaunchUntrackedThread(foo func()) {
 // CallIteratively calls function iteratively in a thread.
 // input param return value is how long to wait before next invocation
 func (s *StopWaiterSafe) CallIteratively(foo func(context.Context) time.Duration) error {
-	return s.CallIterativelyWithTrigger(foo, nil)
+	return s.CallIterativelyWithTrigger(foo, nil, 1)
 }
 
 // CallIterativelyWithTrigger calls function iteratively in a thread.
 // The return value of foo is how long to wait before next invocation
-// Anything sent to triggerChan parameter triggers call to happen immediately
-func (s *StopWaiterSafe) CallIterativelyWithTrigger(foo func(context.Context) time.Duration, triggerChan chan interface{}) error {
+// Anything sent to triggerChan parameter triggers call to happen immediately,
+// but no more than triggerAllowFactor times within normal interval
+func (s *StopWaiterSafe) CallIterativelyWithTrigger(foo func(context.Context) time.Duration, triggerChan <-chan struct{}, triggerAllowFactor uint) error {
```

Comment:
I'm a little conflicted if this belongs in StopWaiter or outside (if it comes from user - the filter should be between user input and StopWaiter).
To be in the StopWaiter it should at least have a disabled setting (e.g. if allowedFactor=0 triggerChan always triggers).
If possible - we'd want that factor to be hot-configurable.. maybe have a function that returns an int instead of an int?